pkg/query-service/app/metrics/v4/helpers/clauses.go

```diff
@@ -38,6 +41,7 @@ func GroupByAttributeKeyTags(tags ...v3.AttributeKey) string {
 func OrderByAttributeKeyTags(items []v3.OrderBy, tags []v3.AttributeKey) string {
 	var orderBy []string
 	for _, tag := range tags {
+		tag.Key = utils.AddBackTickToFormatTag(tag.Key)
 		found := false
 		for _, item := range items {
 			if item.ColumnName == tag.Key {
```

Comment:
since this is replacing `tag.Key` with a new value enclosed in quotes, any subsequent use of `tag.Key` is prone to bugs. For example, in line 47, the comparison doesn't work. Say the passed column name is `service.name` and the new `tag.Key` is "\`service.name\`". Try using order by some query and test it yourself.

Reply:
ok ill check

Reply:
<img width="1512" alt="Screenshot 2024-12-30 at 9 31 58 AM" src="https://github.com/user-attachments/assets/550797e5-3739-491f-a499-10840060ab0e" />
bro i tried, its working for me, is there any particular case that i have missed

Reply:
When you say it's working, you should to provide the evidence to reviewier why it works and whether it is really doing what is expected to do.

Reply:
Don't forget to bring milk :)

Reply:
got he point, need to update the item.columnname, and if !found then tag.key needs to be updated.

Reply:
Since your changes are limited to backend only, the first step in testing should involve looking at the query generated and verify it's correctness and the correctness of data. Then do some sanity testing from the UI.

Reply:
yeah i missed the order by cases,

Reply:
updated the pr please check,